src/main/java/com/artipie/settings/Settings.java

```diff
@@ -95,6 +102,11 @@ final class Fake implements Settings {
          */
         private final Layout layout;
 
+        /**
+         * KeyStore.
+         */
+        private final KeyStore keystore;
```

Comment:
I don't like `Optional` as a field type or as a method parameter.
`Optional` is not a silver bullet to avoid nulls, `Optional` reference itself can be `null`.

I like the `NullObject` pattern:
Everything that can be `null` has a null implementation.
For example, `EmptyKeyStore` that does nothing. So field cannot be `null`, it should be checked at initialization stage.

Reply:
I like the direction, but it seem to be an overhead for this case: this `Fake` implementation can simply throw `NotImplementedException` for keystore-related methods as they are actually not used anywhere

Reply:
Sorry, I didn't understand your thought.
I'd like to note—it's appliable, if implement this [suggestion](https://github.com/artipie/artipie/pull/1160#discussion_r984392259).

Reply:
`Fake` implementation of `Settings` is used for tests only and new keystore related methods are not actually called from any place of code. So we can simply throw `NotImplementedException` in this `Fake` implementation

Reply:
`NullObject` is not a fake or stub implementation.
`NullObject` is one of regular implementations to use in code base.

Reply:
I'm not saying that, check the class or the place (it's `Fake` implementation of `Settings` interface) we are talking about

Reply:
`private final KeyStore keystore` is a field of `Fake` implementation of `Settings` interface, we can avoid adding this field at all, because it actually never used (key-store related methods of `Fake` implementation are never called)

Reply:
Do you mean this field should be removed at all?

Reply:
For now it has Optional type

Reply:
For now it has Optional type

Reply:
yes, the field can be removed and we can throw `NotImplementedException` inside key-store related methods